<a href="https://colab.research.google.com/github/MaximTislenko/GeekBrains/blob/main/26_01_2023.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd
import numpy as np

In [4]:
data_first = pd.read_excel('/content/drive/MyDrive/Наставник/data.xlsx')
data_first

Исполнитель                               Пациент  \
0     Тисленко Максим Витальевич  510DDBB6-73E0-11E5-0D63-485AB665A792   
1     Тисленко Максим Витальевич  510DDBB6-73E0-11E5-0D63-485AB665A792   
2     Тисленко Максим Витальевич  FCDC0ADA-F1BE-40C8-84EB-E13EC4399C75   
3     Тисленко Максим Витальевич  8876df28-dd21-11e9-83cc-00155d7a8301   
4     Тисленко Максим Витальевич  2C6F4DF2-B9D1-11E9-839E-00155D7A8301   
...                          ...                                   ...   
1540  Тисленко Максим Витальевич  42A9FB92-0D66-11EB-1A0D-00155D422D03   
1541  Тисленко Максим Витальевич  F2999F67-4B9C-4EF6-88A0-ED9D90251394   
1542  Тисленко Максим Витальевич  3dd83e60-0e5b-11ed-451b-024279d26107   
1543  Тисленко Максим Витальевич  42A9FB92-0D66-11EB-1A0D-00155D422D03   
1544  Тисленко Максим Витальевич  97bede38-4cb4-11ea-845a-00155d7a8301   

            ДатаРождения Пол  \
0     05.07.2013 0:00:00   М   
1     05.07.2013 0:00:00   М   
2     16.01.2013 0:00:00   Ж   
3     15.04.2011 0:00:00   М   
4     19.07.2019 0:00:00   М   
...                  ...  ..   
1540  13.07.2016 0:00:00   М   
1541  07.09.2006 0:00:00   М   
1542  16.02.2013 0:00:00   М   
1543  13.07.2016 0:00:00   М   
1544  26.12.2018 0:00:00   Ж   

                                           Номенклатура       ДатаВыполнения  \
0     Приём врача-педиатра лечебно-диагностический, ...  03.04.2020 13:20:00   
1     Приём врача-педиатра лечебно-диагностический, ...  03.04.2020 13:20:00   
2     Приём врача-педиатра лечебно-диагностический, ...  10.04.2020 13:30:00   
3     Приём врача-педиатра лечебно-диагностический, ...  10.04.2020 14:00:00   
4     Приём врача-педиатра лечебно-диагностический, ...  11.04.2020 10:00:00   
...                                                 ...                  ...   
1540  Приём врача-педиатра лечебно-диагностический, ...  26.04.2022 18:45:00   
1541  Приём врача-педиатра лечебно-диагностический, ...   19.07.2022 9:00:00   
1542  Приём врача-педиатра лечебно-диагностический, ...  30.07.2022 15:00:00   
1543  Приём врача-педиатра лечебно-диагностический, ...  02.08.2022 16:30:00   
1544  Приём врача-педиатра лечебно-диагностический, ...  18.09.2022 13:15:00   

     МКБ10Код    ТипДиагноза  
0       B82.9  Сопутствующий  
1         J02  Сопутствующий  
2         NaN            NaN  
3         NaN            NaN  
4         NaN            NaN  
...       ...            ...  
1540    K63.9       Основной  
1541      NaN            NaN  
1542      NaN            NaN  
1543      J00       Основной  
1544    J20.9       Основной  

[1545 rows x 8 columns]

In [5]:
data_first.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1545 entries, 0 to 1544
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Исполнитель     1545 non-null   object
 1   Пациент         1545 non-null   object
 2   ДатаРождения    1545 non-null   object
 3   Пол             1543 non-null   object
 4   Номенклатура    1545 non-null   object
 5   ДатаВыполнения  1545 non-null   object
 6   МКБ10Код        1101 non-null   object
 7   ТипДиагноза     1101 non-null   object
dtypes: object(8)
memory usage: 96.7+ KB


###В работу взята выборка по приемам врача-педиатра Тисленко М.В.Данная выписка содержит 1545 записей(строк), за период с 03.04.2020 по 18.09.2022. Для дальнейшего анализа производится переименовывание столбцов, и начинается работа с данными.

In [6]:
data = data_first.copy()
data.rename(columns= {'Исполнитель':'physician',
                      'Пациент':'id_patient',
                      'ДатаРождения':'date_of_birth',
                      'Пол':'Sex',
                      'Номенклатура':'service',
                      'ДатаВыполнения':'visit_date',
                      'МКБ10Код':'ICD_10',
                      'ТипДиагноза':'diagnosis_type'},
             inplace= True)

###Переформатируется в формат дата-время с созданием нового столбцов: Дата рождения, Дата выполнения, для подсчета возраста в день обращения. 

In [8]:
data['birthday'] = pd.to_datetime(data['date_of_birth'], errors= 'coerce', dayfirst= True)
data['visit_date_time'] = pd.to_datetime(data['visit_date'], errors= 'coerce', dayfirst= True)
data['age'] = (data['visit_date_time'] - data['birthday']) // np.timedelta64 (12, 'M')

Добавляется новый столбец для присваивания номера каждому пациенту.

In [9]:
c = data['id_patient'].unique()
index_x = (1 + i for i in range( len(c)))
index_x = pd.DataFrame(index_x)
index_x.columns = ['id']
pation = pd.DataFrame(c)
pation.columns = ['id_patient']
pation = pd.concat([pation,index_x], sort= False, axis=1)
data = data.join(pation.set_index('id_patient'), on= 'id_patient')

In [10]:
data

physician                            id_patient  \
0     Тисленко Максим Витальевич  510DDBB6-73E0-11E5-0D63-485AB665A792   
1     Тисленко Максим Витальевич  510DDBB6-73E0-11E5-0D63-485AB665A792   
2     Тисленко Максим Витальевич  FCDC0ADA-F1BE-40C8-84EB-E13EC4399C75   
3     Тисленко Максим Витальевич  8876df28-dd21-11e9-83cc-00155d7a8301   
4     Тисленко Максим Витальевич  2C6F4DF2-B9D1-11E9-839E-00155D7A8301   
...                          ...                                   ...   
1540  Тисленко Максим Витальевич  42A9FB92-0D66-11EB-1A0D-00155D422D03   
1541  Тисленко Максим Витальевич  F2999F67-4B9C-4EF6-88A0-ED9D90251394   
1542  Тисленко Максим Витальевич  3dd83e60-0e5b-11ed-451b-024279d26107   
1543  Тисленко Максим Витальевич  42A9FB92-0D66-11EB-1A0D-00155D422D03   
1544  Тисленко Максим Витальевич  97bede38-4cb4-11ea-845a-00155d7a8301   

           date_of_birth Sex  \
0     05.07.2013 0:00:00   М   
1     05.07.2013 0:00:00   М   
2     16.01.2013 0:00:00   Ж   
3     15.04.2011 0:00:00   М   
4     19.07.2019 0:00:00   М   
...                  ...  ..   
1540  13.07.2016 0:00:00   М   
1541  07.09.2006 0:00:00   М   
1542  16.02.2013 0:00:00   М   
1543  13.07.2016 0:00:00   М   
1544  26.12.2018 0:00:00   Ж   

                                                service           visit_date  \
0     Приём врача-педиатра лечебно-диагностический, ...  03.04.2020 13:20:00   
1     Приём врача-педиатра лечебно-диагностический, ...  03.04.2020 13:20:00   
2     Приём врача-педиатра лечебно-диагностический, ...  10.04.2020 13:30:00   
3     Приём врача-педиатра лечебно-диагностический, ...  10.04.2020 14:00:00   
4     Приём врача-педиатра лечебно-диагностический, ...  11.04.2020 10:00:00   
...                                                 ...                  ...   
1540  Приём врача-педиатра лечебно-диагностический, ...  26.04.2022 18:45:00   
1541  Приём врача-педиатра лечебно-диагностический, ...   19.07.2022 9:00:00   
1542  Приём врача-педиатра лечебно-диагностический, ...  30.07.2022 15:00:00   
1543  Приём врача-педиатра лечебно-диагностический, ...  02.08.2022 16:30:00   
1544  Приём врача-педиатра лечебно-диагностический, ...  18.09.2022 13:15:00   

     ICD_10 diagnosis_type   birthday     visit_date_time  age   id  
0     B82.9  Сопутствующий 2013-07-05 2020-04-03 13:20:00    6    1  
1       J02  Сопутствующий 2013-07-05 2020-04-03 13:20:00    6    1  
2       NaN            NaN 2013-01-16 2020-04-10 13:30:00    7    2  
3       NaN            NaN 2011-04-15 2020-04-10 14:00:00    8    3  
4       NaN            NaN 2019-07-19 2020-04-11 10:00:00    0    4  
...     ...            ...        ...                 ...  ...  ...  
1540  K63.9       Основной 2016-07-13 2022-04-26 18:45:00    5  159  
1541    NaN            NaN 2006-09-07 2022-07-19 09:00:00   15  336  
1542    NaN            NaN 2013-02-16 2022-07-30 15:00:00    9  661  
1543    J00       Основной 2016-07-13 2022-08-02 16:30:00    6  159  
1544  J20.9       Основной 2018-12-26 2022-09-18 13:15:00    3   60  

[1545 rows x 12 columns]

###На данном этапе уже есть некторые интересные моменты

In [11]:
frequently_ill_children = data['id'].value_counts()
frequently_ill_children[0:50]

53     20
341    16
159    15
21     14
58     13
60     13
158    12
31     12
48     12
25     11
152    11
286    10
27     10
333    10
12     10
103    10
94     10
125    10
357     9
284     9
213     9
164     9
118     9
132     9
7       9
355     8
474     8
156     8
126     8
339     8
490     8
36      8
67      7
211     7
463     7
42      7
26      7
288     7
33      7
32      7
105     7
109     7
17      7
128     7
139     7
204     7
382     6
73      6
178     6
170     6
Name: id, dtype: int64

###В данной таблице представлены Id пациентов, которые чаще всего посещали педиатра. Max - 20 посещений. Если посмотреть возраст этих пациентов это дети в возрасте от 3 до 5 лет